### PR TITLE
Changelog for v2.166.0

### PR DESCRIPTION
**Pull request by vEpiphyte**

### CODE REVIEW

File: CHANGELOG.rst [hunk 1]
```diff
@@ -6,6 +6,131 @@
...
+  (`#3661 <https://github.com/vertexproject/synapse/pull/3661>`_)
+- Add a ``pool`` option to Cron jobs. This can be set to True to enable a
+  Cron job storm query to be executed on a Storm pool member.
+  (`#3652 <https://github.com/vertexproject/synapse/pull/3652>`_)
+- Add a ``pool`` option to Extended HTTP API handlers. This can be set to
+  true to enable an HTTP response to be executed on a Storm pool member.
+  (`#3663 <https://github.com/vertexproject/synapse/pull/3663>`_)
+  (`#3667 <https://github.com/vertexproject/synapse/pull/3667>`_)
+- Add a new Storm API, ``$lib.cortex.httpapi.getByPath()``, that can be
+  used to get a ``http:api`` object by its path. The ``path`` value is
```
Comment (MichaelSquires) [inline CHANGELOG.rst:65]:
```suggestion
  used to get an ``http:api`` object by its path. The ``path`` value is
```

File: CHANGELOG.rst [hunk 2]
```diff
@@ -6,6 +6,131 @@
...
+  evaluated in the same order that the HTTP endpoint resolves the handlers.
+  (`#3663 <https://github.com/vertexproject/synapse/pull/3663>`_)
+- Add ``--list`` and ``--gate`` options to `synapse.tools.modrole`` and
```
Comment (MichaelSquires) [inline CHANGELOG.rst:68]:
```suggestion
- Add ``--list`` and ``--gate`` options to ``synapse.tools.modrole`` and
```

File: CHANGELOG.rst [hunk 3]
```diff
@@ -6,6 +6,131 @@
...
+  ``synapse.tools.moduser``.
+  (`#3632 <https://github.com/vertexproject/synapse/pull/3632>`_)
+- Add a ``view.getMergingViews()`` Storm API. This returns a list of view
+  idens that have open merge requests on view.
+  (`#3666 <https://github.com/vertexproject/synapse/pull/3666>`_)
+- The Storm API ``show:storage`` option now includes storage information for
+  any embedded properties
```
Comment (MichaelSquires) [inline CHANGELOG.rst:75]:
```suggestion
  any embedded properties.
```

File: CHANGELOG.rst [hunk 4]
```diff
@@ -6,6 +6,131 @@
...
+  (`#3656 <https://github.com/vertexproject/synapse/pull/3656>`_)
+- Update the ``LinkShutDown`` exception that a Telepath client may raise to
+  indicate that the connection has been disconnected.
+  (`#3640 <https://github.com/vertexproject/synapse/pull/3640>`_)
+- Add a repr functions for printing the ``aha:pool`` and ``http:api`` objects
```
Comment (MichaelSquires) [inline CHANGELOG.rst:80]:
```suggestion
- Add repr functions for printing the ``aha:pool`` and ``http:api`` objects
```

File: CHANGELOG.rst [hunk 5]
```diff
@@ -6,6 +6,131 @@
...
+  (`#3655 <https://github.com/vertexproject/synapse/pull/3655>`_)
+- Constrain the ``stix2-validator`` library to ``3.0.0,<3.2.0`` due to
+  an API change. This constraint is expected be changed in the next
+  release.
+  (`#3669 <https://github.com/vertexproject/synapse/pull/3669>`_)
+
+Bugfixes
+--------
+- Fix a bug where a Cortex ``promote()`` call could hang when tearing down
+  any running Cron jobs. Cron jobs cancelled during a promotion even will
```
Comment (MichaelSquires) [inline CHANGELOG.rst:99]:
```suggestion
  any running Cron jobs. Cron jobs cancelled during a promotion will
```

File: CHANGELOG.rst [hunk 6]
```diff
@@ -6,6 +6,131 @@
...
+  ``methods`` attribute.
+  (`#3663 <https://github.com/vertexproject/synapse/pull/3663>`_)
+
+Deprecations
+------------
+- The Telepath ``task:init`` message format has been marked as deprecated and
+  will be removed in ``v3.0.0``. This should not affect any users using Synapse
+  ``v2.x.x`` in their client code.
+  (`#3640 <https://github.com/vertexproject/synapse/pull/3640>`_)
+- The authgate with the name ``iden`` is not used for permission checking and
```
Comment (MichaelSquires) [inline CHANGELOG.rst:128]:
```suggestion
- The authgate with the name ``cortex`` is not used for permission checking and
```

File: CHANGELOG.rst [hunk 7]
```diff
@@ -6,6 +6,131 @@
...
+=====================
+
+Automatic Migrations
+--------------------
+- Re-normalize ``risk:mitigation:name``, ``it:mitre:attack:technique:name``,
+  and ``it:mitre:attack:mitigation:name`` secondary properties.
+  (`#3585 <https://github.com/vertexproject/synapse/pull/3585>`_)
+- Re-normalize ``velocity`` properties which are float values.
+  (`#3616 <https://github.com/vertexproject/synapse/pull/3616>`_)
+- See :ref:`datamigration` for more information about automatic migrations.
```
Comment (Cisphyx) [inline CHANGELOG.rst:19]: These were in v2.165.0

File: CHANGELOG.rst [hunk 8]
```diff
@@ -6,6 +6,131 @@
...
+  The following forms had column display hints added to them:
+
+    ``ou:campaign``
+    ``ou:conference``
+    ``ou:goal``
+    ``ou:org``
+    ``ou:team``
+    ``ou:technique``
+    ``ps:contact``
+    ``ps:skil``
```
Comment (Cisphyx) [inline CHANGELOG.rst:37]:
```suggestion
    ``ps:skill``
```

File: CHANGELOG.rst [hunk 9]
```diff
@@ -6,6 +6,131 @@
...
+  (`#3632 <https://github.com/vertexproject/synapse/pull/3632>`_)
+- Add a ``$lib.cache.fixed()`` Storm API. This can be used to get a
+  ``cache:fixed`` object. This cache will execute user provided callbacks
+  written in Storm upon a cache miss.
+  (`#3661 <https://github.com/vertexproject/synapse/pull/3661>`_)
+- Add a ``pool`` option to Cron jobs. This can be set to True to enable a
+  Cron job storm query to be executed on a Storm pool member.
+  (`#3652 <https://github.com/vertexproject/synapse/pull/3652>`_)
+- Add a ``pool`` option to Extended HTTP API handlers. This can be set to
+  true to enable an HTTP response to be executed on a Storm pool member.
```
Comment (Cisphyx) [inline CHANGELOG.rst:61]:
```suggestion
  true to enable an HTTP request handler to be executed on a Storm pool member.
```

File: CHANGELOG.rst (hunk 1 above)
Comment (Cisphyx) [inline CHANGELOG.rst:65]:
```suggestion
  used to get an ``http:api`` object by its path. The ``path`` value is
```

File: CHANGELOG.rst (hunk 2 above)
Comment (Cisphyx) [inline CHANGELOG.rst:68]:
```suggestion
- Add ``--list`` and ``--gate`` options to ``synapse.tools.modrole`` and
```

File: CHANGELOG.rst [hunk 10]
```diff
@@ -6,6 +6,131 @@
...
+  (`#3667 <https://github.com/vertexproject/synapse/pull/3667>`_)
+- Add a new Storm API, ``$lib.cortex.httpapi.getByPath()``, that can be
+  used to get a ``http:api`` object by its path. The ``path`` value is
+  evaluated in the same order that the HTTP endpoint resolves the handlers.
+  (`#3663 <https://github.com/vertexproject/synapse/pull/3663>`_)
+- Add ``--list`` and ``--gate`` options to `synapse.tools.modrole`` and
+  ``synapse.tools.moduser``.
+  (`#3632 <https://github.com/vertexproject/synapse/pull/3632>`_)
+- Add a ``view.getMergingViews()`` Storm API. This returns a list of view
+  idens that have open merge requests on view.
```
Comment (Cisphyx) [inline CHANGELOG.rst:72]:
```suggestion
  idens that have open merge requests on a view.
```

File: CHANGELOG.rst (hunk 5 above)
Comment (Cisphyx) [inline CHANGELOG.rst:99]:
```suggestion
  any running Cron jobs. Cron jobs cancelled during a promotion event will
```

File: CHANGELOG.rst [hunk 11]
```diff
@@ -6,6 +6,131 @@
...
+  blocked until the Storm pool is ready to service requests.
+  (`#3662 <https://github.com/vertexproject/synapse/pull/3662>`_)
+- Ensure that the URL argument provided to ``cortex.storm.pool.set`` can be
+  parsed as a Telepath URL. Previously any string input was accepted.
+  (`#3665 <https://github.com/vertexproject/synapse/pull/3665>`_)
+
+Improved Documentation
+----------------------
+- Update the list of Cortex permissions in the Admin Guide to include
+  ``service.add``, ``service.dell``, ``service.get``,  and ``service.list``.
```
Comment (Cisphyx) [inline CHANGELOG.rst:113]:
```suggestion
  ``service.add``, ``service.del``, ``service.get``,  and ``service.list``.
```

File: CHANGELOG.rst (hunk 6 above)
Comment (Cisphyx) [inline CHANGELOG.rst:128]:
```suggestion
- The authgate with the name ``cortex`` is not used for permission checking and
```

File: CHANGELOG.rst [hunk 12]
```diff
@@ -6,6 +6,122 @@
...
+  (`#3632 <https://github.com/vertexproject/synapse/pull/3632>`_)
+- Add a ``$lib.cache.fixed()`` Storm API. This can be used to get a
+  ``cache:fixed`` object. This cache will execute user provided callbacks
+  written in Storm upon a cache miss.
+  (`#3661 <https://github.com/vertexproject/synapse/pull/3661>`_)
+- Add a ``pool`` option to Cron jobs. This can be set to True to enable a
+  Cron job storm query to be executed on a Storm pool member.
+  (`#3652 <https://github.com/vertexproject/synapse/pull/3652>`_)
+- Add a ``pool`` option to Extended HTTP API handlers. This can be set to
+  true to enable an HTTP request handler to be executed on a Storm pool member.
```
Comment (MichaelSquires) [inline CHANGELOG.rst:52]:
```suggestion
  True to enable an HTTP request handler to be executed on a Storm pool member.
```

**Review by MichaelSquires — DISMISSED** [review]

File: CHANGELOG.rst [hunk 13]
```diff
@@ -6,6 +6,177 @@
...
+  ``inet:tls:clientcert``
+    An x509 certificate sent by a client for TLS.
+
+  **New Properties**
+
+  ``risk:extortion``
+    The form had the following property added to it:
+
+    ``deadline``
+  The time that the demand must be met.
```
Comment (Cisphyx) [inline CHANGELOG.rst:42]:
```suggestion
    ``deadline``
      The time that the demand must be met.
```

**Review by Cisphyx — APPROVED** [review]

**Pull request merged**